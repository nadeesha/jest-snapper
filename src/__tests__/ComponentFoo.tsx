import * as React from 'react';
import { test, init } from '../index';
import * as PropTypes from 'prop-types';

const ComponentFoo: React.StatelessComponent<any> = props =>
  <div {...props}>
    <span {...props} />
  </div>;

ComponentFoo.propTypes = {
  foobarosa: PropTypes.number,
  foo: PropTypes.string,
  bar: PropTypes.number,
  denver: PropTypes.oneOf(['foo', 'bar']),
  brooklyn: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool
  ]),
  michigan: PropTypes.shape({
    foo: PropTypes.string,
    bar: PropTypes.number,
    denver: PropTypes.oneOf(['foo', 'bar']),
    brooklyn: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.bool
    ])
  }),
  buffalo: PropTypes.arrayOf(PropTypes.number),
  cali: PropTypes.objectOf(PropTypes.element),
  sanfran: PropTypes.node,
};

test('should render', ComponentFoo, {
  props: {
    idaho: 'todolo',
  },
});
