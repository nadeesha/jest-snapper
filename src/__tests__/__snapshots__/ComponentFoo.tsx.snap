// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  bar={1337}
  brooklyn="stringval"
  buffalo={
    Array [
      1337,
      1337,
      1337,
      1337,
      1337,
    ]
  }
  cali={
    Object {
      "one": <span
        className="element"
    >
        anyValue
    </span>,
      "two": <span
        className="element"
    >
        anyValue
    </span>,
    }
  }
  denver="foo"
  foo="stringval"
  idaho="todolo"
  michigan={
    Object {
      "bar": 1337,
      "brooklyn": "stringval",
      "denver": "foo",
      "foo": "stringval",
    }
  }
  sanfran={
    <span
      className="node"
    >
      anyValue
    </span>
  }
>
  <span
    bar={1337}
    brooklyn="stringval"
    buffalo={
      Array [
        1337,
        1337,
        1337,
        1337,
        1337,
      ]
    }
    cali={
      Object {
        "one": <span
          className="element"
      >
          anyValue
      </span>,
        "two": <span
          className="element"
      >
          anyValue
      </span>,
      }
    }
    denver="foo"
    foo="stringval"
    idaho="todolo"
    michigan={
      Object {
        "bar": 1337,
        "brooklyn": "stringval",
        "denver": "foo",
        "foo": "stringval",
      }
    }
    sanfran={
      <span
        className="node"
      >
        anyValue
      </span>
    }
  />
</div>
`;
